[IMP] password_security: remove post-install mode in test

yajo opened the pull request:
Usually tests are more resilient like this, and this part of https://github.com/OCA/server-auth/pull/87 was not necessary.

@Tecnativa TT21381

Commit:
[IMP] password_security: remove post-install mode in test

Usually tests are more resilient like this, and this part of https://github.com/OCA/server-auth/pull/87 was not necessary.

@Tecnativa TT21381

hbrunn reviewed (approved):
I think the wide spread usage of this is caused by a wide spread misunderstanding what the effect of this is and what unit tests should be doing

yajo commented:
Merge?

pedrobaeza commented:
Well, everything is red.

Commit:
[FIX] password_security: tests compatible with auth_totp_password_security

For some weird reason, this mock is not removed properly and makes auth_totp_password_security fail with:


```
2020-02-18 08:59:41,120 5720 INFO openerp_test odoo.addons.auth_totp_password_security.tests.test_main: test_mfa_login_post_pass_not_expired (odoo.addons.auth_totp_password_security.tests.test_main.TestAuthTotpPasswordSecurity)
2020-02-18 08:59:41,120 5720 INFO openerp_test odoo.addons.auth_totp_password_security.tests.test_main: ` Should return result of super if user's password not expired
2020-02-18 08:59:41,134 5720 INFO openerp_test odoo.addons.auth_totp_password_security.tests.test_main: ======================================================================
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: FAIL: test_mfa_login_post_expired_redirect (odoo.addons.auth_totp_password_security.tests.test_main.TestAuthTotpPasswordSecurity)
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: ` Should return correct redirect if password is expired
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: Traceback (most recent call last):
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: `   File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: `     return func(*args, **keywargs)
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: `   File "/home/travis/build/OCA/server-auth/auth_totp_password_security/tests/test_main.py", line 76, in test_mfa_login_post_expired_redirect
2020-02-18 08:59:41,135 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: `     self.assertIn(expected, result)
2020-02-18 08:59:41,136 5720 ERROR openerp_test odoo.addons.auth_totp_password_security.tests.test_main: ` AssertionError: 'http://localhost:8069/web/reset_password?db=openerp_test&token=5puYy7xez6ue7FvYI0iN' not found in 'redirected'
```

I changed the test to avoid the failure. This repo has lots of unnecessary and weird usages of mock and the bug might be elsewhere, but this fixes the branch for now.

@Tecnativa TT21381

pedrobaeza commented:
/ocabot merge
